packages/monosize/src/types.mts

```diff
@@ -46,4 +46,21 @@ export type MonoSizeConfig = {
   repository: string;
   storage: StorageAdapter;
   bundler: BundlerAdapter;
+  /**
+   * Override package name resolution used within compare-reports and upload-report.
+   *
+   * By default we try to read package name from "package.json" or "project.json" files.
+   * You can override this behavior by providing your own implementation.
+   */
+  packageName?: (packageRoot: string) => Promise<string>;
```

Comment:
I have to say these namings are confusing from user POV, should we rather go with something like I originally proposed to express the intent more clearly?

```ts
{
reportResolvers?: {
  packageName: (packageRoot: string) => Promise<string>;
  packageRoot?: (reportFilePath: string) => Promise<string>;
}
}
```

Reply:
Agree, we also would be able to extend it later with `findProjectRoot` / `findMonorepoRoot` as currently our assumption is that `findGitRoot()` is enough, but it might not.

Reply:
updated